miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -62,8 +62,21 @@ abstract class MiniAppMessageBridge {
         permissionsWithDescription: List<Pair<MiniAppCustomPermissionType, String>>,
         callback: (List<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>) -> Unit
     ) {
-        throw MiniAppSdkException("The `MiniAppMessageBridge.requestCustomPermissions`" +
-                " method has not been implemented by the Host App.")
+        throw MiniAppSdkException(
+            "The `MiniAppMessageBridge.requestCustomPermissions`" +
+                    " method has not been implemented by the Host App."
+        )
+    }
+
+    /**
+     * Post user name request from external.
+     * @param callback to return user name.
+     */
+    open fun requestUserName(callback: (isSuccess: Boolean, data: String) -> Unit) {
```

Comment:
I believe we should break down the `MiniAppMessageBridge` here.

Can we delegate `requestUserName` & `requestUserPhoto` into another component class such as `UserInfoBridge`? The `UserInfoBridge` belongs to `MiniAppMessageBridge`.
I will divide other existed functions in the future too because `MiniAppMessageBridge` is too big now. 

@khairul-alam-licon @corycaywood Please feel free to suggest other design patterns to divide `MiniAppMessageBridge` smaller.

Reply:
I was thinking about the same thing, but I haven't thought of a good way of doing this. The class is also getting quite long for the Host App to implement since they need to override so many functions.

Maybe we could add a function for `MiniAppMessageBridge.setUserInfoProvider`? (similar to `MiniAppMessageBridge.setAdMobDisplayer`)

Reply:
Well I think, the purpose of defining `Bridge` postfix means a component to communicate with external MiniApp customers. We may have `UserInfoBridge`, but the name may confuses the users with multiple bridges and the convention will be difficult to manage with lots of features. 
Or, we could write `MiniAppMessageBridgeDelegate` which will be extended by `MiniAppMessageBridge` but it will solve the "longer component" issue temporarily. 

In the other side, logically `UserInfoProvider` is only for providing the data, not for other operations. We can have this class optionally, but it will not solve the actual problem. 

In my opinion, we can have something like-

```
internal class UserInfoHandler(private val bridge: MiniAppMessageBridge) {
  
   fun onRequestUserName(callbackObj: CallbackObj) {

         // get data from UserInfoProvider 
         // implement the usecases of postValue / postError

   }
}
```
And, we can do the same for all the other features to hold the single responsibility. And, in that's way, we can have a single messaging `Bridge` for the users.

Let me know, what is in your mind.

Reply:
Well I think, the purpose of defining `Bridge` postfix means a component to communicate with external MiniApp customers. We may have `UserInfoBridge`, but the name may confuses the users with multiple bridges and the convention will be difficult to manage with lots of features. 
Or, we could write `MiniAppMessageBridgeDelegate` which will be extended by `MiniAppMessageBridge` but it will solve the "longer component" issue temporarily. 

In the other side, logically `UserInfoProvider` is only for providing the data, not for other operations. In my opinion, we can have something like-

```
internal class UserInfoHandler(private val bridge: MiniAppMessageBridge) {
  
   fun onRequestUserName(callbackObj: CallbackObj) {

         // get data from UserInfoProvider 
         // implement the usecases of postValue / postError

   }
}
```

`UserInfoHandler` is an example, we can have alternative postfix like `UserInfoHelper`, `UserInfoRequester` etc.

And, we can do the same for all the other features to hold the single responsibility. 
Let me know, what is in your mind.

Reply:
@khairul-alam-licon I think it's good to go.

Reply:
I have updated the PR with `UserInfoHandler` idea. if it's sound good, I'll forward to get profile-photo. In this PR, it should be fine.